kolibri/utils/server.py

```diff
@@ -444,29 +459,26 @@ def stop():
     if not pid:
         return status
 
-    kill_pid(pid)
-    wait_for_status(STATUS_STOPPED)
-    if pid_exists(pid):
-        logger.debug(
-            "Process wth pid %s still exists after soft kill signal; attempting a SIGKILL."
-            % pid
-        )
-        os.kill(pid, signal.SIGKILL)
-    starttime = time.time()
-    while time.time() - starttime <= 10:
+    with open(COMMAND_FLAG, "w") as f:
+        f.write(STOP)
+    if not wait_for_status(STATUS_STOPPED, timeout=30):
         if pid_exists(pid):
-            time.sleep(0.1)
-        else:
-            break
-    if pid_exists(pid):
-        logger.debug(
-            "Process wth pid %s still exists after soft kill signal; attempting a SIGKILL."
-            % pid
-        )
-        os.kill(pid, signal.SIGKILL)
-    if pid_exists(pid):
-        logging.error("Kolibri process has failed to shutdown")
-        return STATUS_UNCLEAN_SHUTDOWN
+            logger.debug("Process wth pid %s still exists; attempting a SIGKILL." % pid)
+            try:
+                os.kill(pid, signal.SIGKILL)
+            except SystemError as e:
+                logger.debug(
```

Comment:
I'm a little concerned that this could potentially cause reentrancy issues, since calls to `stderr` are reentrant, although I'm not totally certain that it would.

Reply:
The process it is killing is a different process to the one where the SystemError is being thrown, so I don't think this should be a concern.

Reply:
Sounds good!